cloud-gov/cg-ui
# sets up s3 bucket access for last logon, removes mock data for org user list

## Changes proposed in this pull request:

- hooks up to s3 bucket to pull down user information (filters to org users in controller helper)
- org page display toggles between s3 data or not displaying any latest logon info
- pulls out UAA in controller, leaves API code in place in case of future needs
- updates readme to explain how to connect to s3 bucket for local development

Questions:

- ✅ do we want to pull out ALL the UAA API code?
- ✅ are we okay with me removing the manual deploy instructions from the readme?
- ✅ is the filtering for the user information in the controller the best place to do this? We could also set up the s3 API code to act as an API that accepts user ids and returns only those, though they are both happening server side so the preference comes down to where it makes the most sense for us.
- (#330 ) how do we want to distinguish between someone who has never logged in (no entry in json) vs development mode (no json at all) when it gets to the component level?

### How to test

Without s3:  Start up the app locally like normal and go to an org details page (/orgs/[orgId]). You should see a message saying that log in information was unavailable

With s3:  Follow the added instructions in this PR's README to create a service key and add the credentials to your .env.local file. Start up the app, this time you should see log in information for each user.

### Screenshots

When s3 returns an error or is not set up for local development:
![image](https://github.com/cloud-gov/cg-ui/assets/2480492/366483ad-bd05-4955-9e61-37289290551c)

An example of real data from s3, but where a user has never logged in (and therefore isn't being caught by the uaa-bot -- this behavior differs from the previous UAA response which would have returned the user but shown them as inactive):

![image](https://github.com/cloud-gov/cg-ui/assets/2480492/cc1899f1-6bbc-4a91-a47b-139361eef2a3)

### Related issues

closes #307

### Submitter checklist


## Security considerations

This PR uses data passed from UAA to an S3 bucket, but without any of the same mechanisms the UAA API has to ensure that users should have access to the information. We are instead relying on the CAPI API's list of organization users to filter the information appropriately within our application.
